src/pages/Detail/Detail.js

```diff
@@ -0,0 +1,283 @@
+import React, { useState, useRef, useEffect } from 'react';
+import DetailModal from '../../components/DetailModal/DetailModal';
+import './Detail.scss';
+
+const Detail = () => {
+  const [detailContents, setDetailContents] = useState([]);
+  const [number, setNumber] = useState(0);
+  const [isModalOpen, setIsModalOpen] = useState(false); //모달 상태 관리 : 기본값 - 닫힘
+  const dimmerRef = useRef();
+  const [currentSlide, setCurrentSlide] = useState(0);
+  const [currentSlideWrap, setCurrentSlideWrap] = useState(0);
+  const TOTAL_SLIDES = 1;
+  const TOTAL_SLIDES_WRAP = 2;
+  const slideRef = useRef();
+  const slideWrapRef = useRef();
+
+  // useEffect(() => {
+  //   fetch('http://10.58.5.235:8000/products/detail/1')
+  //     .then(response => response.json())
+  //     .then(result => setDetailContents(result));
+  //   //변수를 만들어서 result를 담아줌
+  //   //그 담은 변수에다가 map을 돌려야 함.
+  //   //
+  // }, []);
+  // console.log(detailContents);
+
+  useEffect(() => {
+    // dimmerRef.current.style;
+    // dimmerRef.current.style.display = 'block';
+  }, [isModalOpen]); // 의존, 뭘 의존하냐? 예슬님이 주는 커피 / 조건 /
+
+  const increaseAmount = () => {
+    setNumber(number + 1);
+  };
+
+  const decreaseAmount = () => {
+    if (number === 0) return;
+    setNumber(number - 1);
+  };
```

Comment:
기존의 state를 이용해서 새로운 값을 setState하는 경우에는 함수형 업데이트로 해주세요!
```suggestion
  const increaseAmount = () => {
    setNumber(number => number + 1);
  };

  const decreaseAmount = () => {
    if (number === 0) return;
    setNumber(number => number - 1);
  };
```

Reply:
함수형 업데이트의 장점
- 비동기 상황에서 제일 최신의 prevState 값이 보장됨 (number가 closure가 아니기 때문)
- 그렇기 때문에 scope에서 좀 더 안전하다 (number라는 변수에 다른 값이 할당됐을 수도 있기 때문)
- 따라서 이전 state를 이용해서 새로운 state를 setState할 때는 함수형 업데이트로 하는 게 안정적!

Reply:
수정 후 더 공부해보겠습니다 감사합니다 👍